### DISCUSSION TITLE
Custom accessor in addition to foreign key

### DISCUSSION THREAD

**Discussion by dimamik** [Adrinalin4ik/Nestjs-Graphql-Tools#9] [Ideas]
It would be great to be able to do something like:

```ts
  @GraphqlLoader({
    accessor: (task: Task) => task.chunk.id,
  })
```

**Comment by Adrinalin4ik** [comment]
WoW, it looks good! Thank you for the suggestion, I'll take a look what I can do:)

> It would be great to be able to do something like:
> 
> ```ts
>   @GraphqlLoader({
>     accessor: (task: Task) => task.chunk.id,
>   })
> ```

WoW, it looks good! Thank you for the suggestion, I'll take a look what I can do:)

**Comment by dimamik** [comment]
I've introduced it in this fork for our needs, so it should be pretty straightforward. 

https://github.com/agh-kiwis/agh-kiwis/blob/82b7951493a407984ca6684e738f282a1405bc5d/libs/nestjs-graphql-tools/src/loader.ts#L211-L216

**Comment by Adrinalin4ik** [comment]
Btw, @Graphql has foreignKey parameter. Does it work for you?

**Comment by dimamik** [comment]
@Adrinalin4ik 
Nope, as I need to deal with nested relations.

**Comment by Adrinalin4ik** [comment]
Got it, will do that tomorrow.

**Comment by Adrinalin4ik** [comment]
PR https://github.com/Adrinalin4ik/Nestjs-Graphql-Tools/pull/10

**Reply by Adrinalin4ik** [reply]
version 0.7.4
